### ISSUE TITLE
Activate core packages before activating third party packages

### ISSUE THREAD

**Issue by steelbrain** [atom/atom#9522]
In package activation, `linter-*` is activated **before** `notifications` package. Which results in https://github.com/steelbrain/package-deps/issues/25
I have to work it around by waiting 20 ms before getting the view of the notification, see [this line](https://github.com/steelbrain/package-deps/pull/26/files#diff-2ee7035f560888fa5c0f1c0e929032fdR21)

It would be nice to have this issue resolved in the core instead of working it around in every package.

**Comment by benogle** [comment]
This kind of doesnt make sense. It looks like something in linter-perl is calling `atom.views.getView` on a notification. Why?

**Comment by steelbrain** [comment]
@benogle [here's the line](https://github.com/steelbrain/package-deps/pull/26/files#diff-2ee7035f560888fa5c0f1c0e929032fdR23) that calls `atom.views.getView` on the notification, it does this so we can "monkey patch" a few things in it, and change the info notification into a success notification once the installation is complete.

But that's not the problem here, because the notification registry knows it's an instance of the `Notification` class and still throws that error, it needs to be fixed :)

**Comment by benogle** [comment]
Maybe the solution is to add something to the notification that can handle the progress case. Ideally we'd avoid the need for monkey patches like this, and avoid reliance on order of operations.

We _could_ register the notification view outside of the activation method, but that could have an impact on startup speed.

**Comment by steelbrain** [comment]
@benogle I do agree that is the long term fix, but it would take a lot of time, I figured it would be a faster way to monkey patch it in my package, but I'll still be happy with it if you guys choose to implement that instead of activation of core packages before third party ones

**Comment by benogle** [comment]
Can you experiment with registering the NotificationView outside of the activate method? I am down if it does not add anything to startup time. That would be the most minimal code change.

**Comment by steelbrain** [comment]
@benogle I will do that and report back, Thanks!

**Comment by lee-dohm** [comment]
@steelbrain Is there any update on this? Or can this be closed?

**Comment by steelbrain** [comment]
Sorry @lee-dohm but I've been *very* busy lately and haven't been able to work on this. The issue still remains. I have a question in mind however, @benogle isn't the module's main file only included when the module is about to be activated? Why would it make a difference if we register the view outside of activate. We will still run into this race-condition where third party packages try to get view of elements that core packages haven't registered yet.

**Comment by lee-dohm** [comment]
Thanks for the update @steelbrain! It's quite alright, I just wanted to confirm :grinning:

**Comment by steelbrain** [comment]
Potential fix is in https://github.com/atom/atom/pull/10891
